backend/user-service/src/controllers/user.controller.ts

```diff
@@ -32,3 +34,28 @@ export async function handleCreateUser(request: TypedRequest<CreateUserDto>, res
 
     response.status(201).json(dto).send()
 }
+
+export async function handleUpdateProfile(request: TypedRequest<UserProfileDto>, response: Response): Promise<void> {
+    const createDto = UserProfileDto.fromRequest(request)
+    const errors = await createDto.validate()
+    if (errors.length) {
+        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
+        response.status(400).json(errorMessages).send()
+        return
+    }
+
+    const id = request.params.id
+
+    const duplicateUsername = await findOneUserByUsername(createDto.username)
+    if (duplicateUsername && duplicateUsername.id !== id) {
+        response.status(409).json(['DUPLICATE_USERNAME']).send()
+        return
+    }
+
+    try {
+        const user = await updateUser(id, createDto)
+        response.status(200).json(user).send()
+    } catch (e) {
+        response.status(404).json([e]).send()
```

Comment:
If the update method throws errors, it should be an internal error, maybe something wrong with the state of the db. In this case, it may be more appropriate to return a 500. The error also should not be exposed to the client. Perhaps you could use the logger to log the error.

Reply:
You can also remove the throw, the default error handler in index.ts should catch any errors. However, mongoose async errors are abit buggy, might need to test out if u let it propagate up to the root error handler.

Reply:
my initial approach was without the try-catch, but the tests would wait with a timeout error for cases where the id doesnt exist in the table. that's why, i added the try-catch block.

i agree that the error should not be exposed to the client, but when i dont use the e, the lint throws an error lmao...

Reply:
interesting, so thats the reason for the try catch. Why don't you just check if the user exists first by doing a query by id?

Reply:
yeah that could work as well, but it would make it a bit more verbose and would require another query.. 

lemme know if you want me to go for that approach, and ill change the code

Reply:
this is fine for now, lgtm